File: Frontend/src/app/system/_domains/Supplier.ts
import { Injectable } from '@angular/core';
import { _EntityBase } from './_EntityBase';
import { LegalPerson } from './LegalPerson';
import { SupplierType } from './enums/SupplierType';

@Injectable()
export class Supplier extends _EntityBase {
     public IdLegalPerson: number;
     public FKLegalPerson: LegalPerson;
     public MoreInformation: string;
     public SupplierType: SupplierType;
}

File: Backend/Stock.Domain/Supplier.cs
using Core.Domain;
using Core.Domain.Enum;
using Framework.Domain;
using Framework.Domain.Attributes;
using Stock.Domain.Enum;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Stock.Domain
{
    public class Supplier : EntityBase
    {
        [Required(ErrorMessage = "A pessoa é obrigatoria")]
        public int IdLegalPerson { get; set; }
        [NotMapped]
        [Filter(true)]
        public LegalPerson FKLegalPerson { get; set; }
        [Filter("FKLegalPerson.Name")]
        public string FKLegalPersonName { get { return FKLegalPerson != null ? FKLegalPerson.Name : ""; } }

        public string MoreInformation { get; set; }

        public SupplierType SupplierType { get; set; }
    }
}
